lib/src/results.dart

```diff
@@ -110,11 +127,21 @@ extension RealmResultsInternal on RealmResults {
     return _handle;
   }
 
-  RealmObjectMetadata? get metadata => _metadata;
-
-  static RealmResults<T> create<T extends RealmObjectBase>(RealmResultsHandle handle, Realm realm, RealmObjectMetadata? metadata) {
-    return RealmResults<T>._(handle, realm, metadata);
-  }
+  RealmObjectMetadata get metadata => _metadata!;
+
+  static RealmResults<T> create<T extends Object?>(
+    RealmResultsHandle handle,
+    Realm realm,
+    RealmObjectMetadata? metadata,
+  ) =>
+      RealmResults<T>._(handle, realm, metadata);
+
+  static RealmResults<T> createFromList<T extends Object?>(
+    RealmList list,
+    Realm realm,
+    RealmObjectMetadata? metadata,
+  ) =>
+      RealmResults<T>._(realmCore.resultsFromList(list), realm, metadata);
```

Comment:
It's kind of weird this is here. Why not have `list.asResults` call `realmCore.resultsFromList(list)` and then use the handle to call `ResultsInternal.create`?

Reply:
good point. essentially removing `createFromList`

Reply:
Fixed